### ISSUE TITLE
New Rule: SelectorSortOrder

### ISSUE THREAD

**Issue by nschonni** [brigade/scss-lint#112]
The rule would ensure that the selectors would be sorted by their CSS specificity (elements/pseodos, classes, IDs).

**Bad**:

``` scss
.foo {
  color: #fff;

  .bar {
    color: #00f;
  }

  p {
    color: #f00;
  }

  #baz {
    color: #0f0;
  }
}
```

**Good**:

``` scss
.foo {
  color: #fff;

  p {
    color: #f00;
  }

  .bar {
    color: #00f;
  }

  #baz {
    color: #0f0;
  }
}
```

Not sure if this would get too complicated with compound selectors
